src/Symfony/Component/Routing/RouteCollection.php

```diff
@@ -116,6 +126,10 @@ public function all()
      */
     public function get(string $name)
     {
+        while ($this->hasAlias($name)) {
+            $name = $this->getAlias($name)->getId();
```

Comment:
shouldn't this loop detect circular refs + trigger a deprecation?
why do we need this logic btw? don't we always resolve the aliases before calling the method?
alternatively, shouldn't we move the logic in CompiledUrlGeneratorDumper and UrlGenerator here?